### PR TITLE
feat(ex/phoenix): Upgrade Phoenix to 1.7

### PR DESCRIPTION
**Pull request by joshlarson**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206113388125707

### CODE REVIEW

File: lib/skate_web.ex [hunk 1]
```diff
@@ -19,7 +19,7 @@ defmodule SkateWeb do
 
   def controller do
     quote do
-      use Phoenix.Controller, namespace: SkateWeb
+      use Phoenix.Controller, namespace: SkateWeb, layouts: [html: {SkateWeb.LayoutView, :app}]
```
Comment (joshlarson) [inline lib/skate_web.ex:22]: Jose Valim said to do this, so it must be right.

File: lib/skate_web/views/layout_view.ex [hunk 2]
```diff
@@ -1,5 +1,6 @@
 defmodule SkateWeb.LayoutView do
-  use SkateWeb, :view
+  use(SkateWeb, :html)
+  use(SkateWeb, :view)
```
Comment (firestack) [inline lib/skate_web/views/layout_view.ex:3]: nitpick: In the past the formatter hasn't forced parentheses on `use` statements, do you know why it's happened here / what changed?

File: assets/package-lock.json [hunk 3]
```diff
@@ -95,11 +95,11 @@
       }
     },
     "../deps/phoenix": {
-      "version": "1.6.16",
+      "version": "1.7.10",
       "license": "MIT"
     },
     "../deps/phoenix_html": {
-      "version": "3.3.1"
+      "version": "3.3.3"
```
Comment (firestack) [inline assets/package-lock.json:102]: https://github.com/mbta/skate/pull/2300#discussion_r1419106151 Seems like we might want to merge the dep update before this?

**Comment by lemald** [comment]
@joshlarson just FYI, if you rebase on the latest `master` the flaky Elixir test should be fixed.

File: lib/skate_web/views/layout_view.ex (hunk 2 above)
Comment (joshlarson) [reply lib/skate_web/views/layout_view.ex:3]: I have absolutely no idea. I assumed it _was_ a formatter thing, which is why I didn't change it back, but I just fixed it and the formatter isn't fighting me.

File: mix.exs
[the remark this answers is not in the document]
Comment (joshlarson) [reply mix.exs:1]: Yeah - if there are any issues, I'll address them in this PR

**Review by firestack — APPROVED** [review]
LGTM

**Review by lemald — APPROVED** [review]
🚀

**Pull request merged**